### PR TITLE
un-i18n Modal Analytics

### PR DESCRIPTION
**Pull request by t3chguy**
Signed-off-by: Michael Telatynski <7t3chguy@gmail.com>

paired with https://github.com/matrix-org/matrix-react-sdk/pull/1255

### CODE REVIEW

**Comment by t3chguy** [comment]
# Await Manual Testing

File: src/components/structures/RoomDirectory.js [hunk 1]
```diff
@@ -327,14 +327,14 @@ module.exports = React.createClass({
...
-                    Modal.createDialog(ErrorDialog, {
+                    Modal.createTrackedDialog('Room not found', '', ErrorDialog, {
                         title: _t('Room not found'),
                         description: _t('Couldn\'t find a matching Matrix room'),
                     });
                 }
             }, (e) => {
                 const ErrorDialog = sdk.getComponent("dialogs.ErrorDialog");
-                Modal.createDialog(ErrorDialog, {
+                Modal.createTrackedDialog('Fetching third party location failed', e.toString(), ErrorDialog, {
```
Comment (lukebarnard1) [inline src/components/structures/RoomDirectory.js]: e.toString()

File: src/components/views/context_menus/MessageContextMenu.js [hunk 2]
```diff
@@ -99,7 +99,7 @@ module.exports = React.createClass({
                     const ErrorDialog = sdk.getComponent("dialogs.ErrorDialog");
                     // display error message stating you couldn't delete this.
                     const code = e.errcode || e.statusCode;
-                    Modal.createDialog(ErrorDialog, {
+                    Modal.createTrackedDialog('You cannot delete this message', e.toString(), ErrorDialog, {
```
Comment (lukebarnard1) [inline src/components/views/context_menus/MessageContextMenu.js]: e.toString()

File: src/components/views/elements/ImageView.js [hunk 3]
```diff
@@ -76,7 +76,7 @@ module.exports = React.createClass({
                     var ErrorDialog = sdk.getComponent("dialogs.ErrorDialog");
                     // display error message stating you couldn't delete this.
                     var code = e.errcode || e.statusCode;
-                    Modal.createDialog(ErrorDialog, {
+                    Modal.createTrackedDialog('You cannot delete this image.', e.toString(), ErrorDialog, {
```
Comment (lukebarnard1) [inline src/components/views/elements/ImageView.js]: e.toString()

File: src/components/views/settings/Notifications.js [hunk 4]
```diff
@@ -131,7 +131,7 @@ module.exports = React.createClass({
             this._refreshFromServer();
         }, (error) => {
             var ErrorDialog = sdk.getComponent("dialogs.ErrorDialog");
-            Modal.createDialog(ErrorDialog, {
+            Modal.createTrackedDialog('Error saving email notification preferences', error.toString(), ErrorDialog, {
```
Comment (lukebarnard1) [inline src/components/views/settings/Notifications.js]: e.toString()

File: src/components/views/settings/Notifications.js [hunk 5]
```diff
@@ -241,7 +241,7 @@ module.exports = React.createClass({
             }, function(error) {
                 var ErrorDialog = sdk.getComponent("dialogs.ErrorDialog");
                 console.error("Failed to change settings: " + error);
-                Modal.createDialog(ErrorDialog, {
+                Modal.createTrackedDialog('Failed to change settings', error.toString(), ErrorDialog, {
```
Comment (lukebarnard1) [inline src/components/views/settings/Notifications.js]: e.toString()

File: src/components/views/settings/Notifications.js [hunk 6]
```diff
@@ -311,7 +311,7 @@ module.exports = React.createClass({
         }, function(error) {
             var ErrorDialog = sdk.getComponent("dialogs.ErrorDialog");
             console.error("Can't update user notification settings: " + error);
-            Modal.createDialog(ErrorDialog, {
+            Modal.createTrackedDialog('Can\'t update user notifcation settings', error.toString(), ErrorDialog, {
```
Comment (lukebarnard1) [inline src/components/views/settings/Notifications.js]: e.toString()

File: src/components/views/settings/Notifications.js [hunk 7]
```diff
@@ -353,7 +353,7 @@ module.exports = React.createClass({
         var onError = function(error) {
             var ErrorDialog = sdk.getComponent("dialogs.ErrorDialog");
             console.error("Failed to update keywords: " + error);
-            Modal.createDialog(ErrorDialog, {
+            Modal.createTrackedDialog('Failed to update keywords', error.toString(), ErrorDialog, {
```
Comment (lukebarnard1) [inline src/components/views/settings/Notifications.js]: e.toString()

**Comment by t3chguy** [comment]
@richvdh Test Flake: https://travis-ci.org/vector-im/riot-web/jobs/263105402

**Pull request merged**